mesh/mesh.go

```diff
@@ -81,45 +85,79 @@ func (m *mesh) AddLayer(layer *Layer) error {
 		return errors.New("can't add layer missing previous layers")
 	}
 
-	m.mDB.addLayer(layer)
+	m.addLayer(layer)
 	m.tortoise.HandleIncomingLayer(layer)
-	atomic.AddUint32(&m.latestIrreversible, 1)
-	m.SetLatestKnownLayer(uint32(layer.Index()))
+	atomic.AddUint32(&m.localLayer, 1)
+	m.SetLatestLayer(uint32(layer.Index()))
 	return nil
 }
 
 func (m *mesh) GetLayer(i LayerID) (*Layer, error) {
 	m.lMutex.RLock()
-	if i > LayerID(m.latestIrreversible) {
+	if i > LayerID(m.localLayer) {
```

Comment:
I don't think there should be a limit on what getlayer should return, should we have an error if the later is not verified? if it's a helper method for this package make it private